### ISSUE TITLE
False positive for platform-conditional sublibraries using (select) and (enabled_if)

### ISSUE THREAD

**Issue by tmcgilchrist** [ocurrent/opam-dune-lint#75]
When a dune project uses `(select)` to conditionally depend on an internal sublibrary that is only built on certain platforms (via `enabled_if`), `opam-dune-lint` reports a false positive, suggesting it should be added as an opam dependency.

## Reproducer

The [ocaml-processor](https://github.com/haesbaert/ocaml-processor) project has platform-specific sublibraries for Apple IORegistry and AMD64 CPUID. The main library conditionally selects them:

```dune
(library
 (name processor)
 (public_name processor)
 (libraries
  unix
  str
  (select
   ioreg.ml
   from
   (processor_apple_ioreg -> ioreg.default.ml)
   (-> ioreg.nop.ml))
  (select
   amd64.ml
   from
   (amd64 -> amd64.default.ml)
   (-> amd64.nop.ml))))
```

The `processor_apple_ioreg` sublibrary is only built on macOS:

```dune
(library
 (name processor_apple_ioreg)
 (public_name processor.apple_ioreg)
 (enabled_if
  (= %{system} "macosx"))
 ...)
```

When `opam-dune-lint` runs on Linux (e.g. in CI), it fails with:

```
WARNING: can't find opam package providing "processor_apple_ioreg"!
processor.opam: changes needed:
  "processor_apple_ioreg" {>= "0"}         [from lib]
```

This is wrong — `processor_apple_ioreg` is an internal sublibrary within the same project, not an external dependency. The `(select)` stanza already handles the case where it's unavailable by falling back to `ioreg.nop.ml`.

## Expected behaviour

`opam-dune-lint` should recognise that `processor_apple_ioreg` is an internal sublibrary (it's defined in the same project with `public_name processor.apple_ioreg`) and not suggest adding it as an opam dependency, even when `enabled_if` prevents it from being built on the current platform.

## Environment

- opam-dune-lint 0.6
- dune 3.2+
- Running on Ubuntu (CI), where the macOS-only sublibrary is not built
